### DISCUSSION TITLE
Help HLS to UDP configuration

### DISCUSSION THREAD

**Discussion by eriktovarj** [bluenviron/mediamtx#2295] [Questions] [answered]
Hi!
Greetings from Mexico.
This is what i want to do: There are multiple HLS links (.m3u8) that I want to change to a UDP format. What configuration do i have to do, to get what I'm looking for?

Do i have to READ the stream m3u8 first and then publish or do i publish directlly from the HLS stream?

Any thinkg will be appreciated as im new. Im willing to pay if needed, as this needs to get done ASAP.

Thanks in advance :D

**Comment by aler9** [comment, marked as answer]
Hello, i think that FFmpeg is enough to perform the conversion:

```
ffmpeg -re -i http://your-hls-url.m3u8 -c copy -f mpegts udp://destination-ip:destination-port?pkt_size=1316
```

**Reply by eriktovarj** [reply]
where can i place all the links i need to convert.
Another question:
What specs does the PC needs to make the conversion of 50 channels?

**Reply by aler9** [reply]
you don't need MediaMTX at all (i know, i wrote MediaMTX but for some use cases it's useless :smile:), install FFmpeg and paste the commands in the terminal. Regarding the specs, the most important thing is the bandwidth of your internet connection, not the PC specification, since a change of protocol usually requires a very low amount of CPU.

**Reply by eriktovarj** [reply]
Bandwith is not an issue, we have a 10G connection.
http://stitcher-ipv4.pluto.tv/v1/stitch/embed/hls/channel/5dcb62e63d4d8f0009f36881/master.m3u8?deviceType=samsung-tvplus&deviceMake=samsung&deviceModel=samsung&deviceVersion=unknown&appVersion=unknown&deviceLat=0&deviceLon=0&deviceDNT=%7BTARGETOPT%7D&deviceId=%7BPSID%7D&advertisingId=%7BPSID%7D&us_privacy=1YNY&samsung_app_domain=%7BAPP_DOMAIN%7D&samsung_app_name=%7BAPP_NAME%7D&profileLimit=&profileFloor=&embedPartner=samsung-tvplus 

The previous link didnt work, it returned the following:

[1]+  Stopped                 ffmpeg -re -i http://stitcher-ipv4.pluto.tv/v1/stitch/embed/hls/channel/5dcb62e63d4d8f0009f36881/master.m3u8?deviceType=samsung-tvplus
[2]   Done                    deviceMake=samsung
[3]   Done                    deviceModel=samsung
[4]   Done                    deviceVersion=unknown
[5]   Done                    appVersion=unknown
[6]   Done                    deviceLat=0
[7]   Done                    deviceLon=0
[8]   Done                    deviceDNT=%7BTARGETOPT%7D
[9]   Done                    deviceId=%7BPSID%7D
[10]   Done                    advertisingId=%7BPSID%7D
[11]   Done                    us_privacy=1YNY
[12]   Done                    samsung_app_domain=%7BAPP_DOMAIN%7D
[13]   Done                    samsung_app_name=%7BAPP_NAME%7D
[14]   Done                    profileLimit=
[15]   Done                    profileFloor=

How do i have to place the link in the example you gave me?
Thank a lot for your help. i would like to contribute in an economic way to your great project

**Reply by aler9** [reply]
you have to put the link within single quotes, otherwise the ampersand will get recognized as a special character:

```
ffmpeg -re -i 'http://your-hls-url.m3u8' -c copy -f mpegts 'udp://destination-ip:destination-port?pkt_size=1316'
```

**Reply by eriktovarj** [reply]
how do i send multiple commands or how do i create a batch file (im more of a windows user) to convert all of them at once.
I beleive your mediaMTX works for me as i can use the API, the performance monitor and the statistics.
i sent you an email to your gmail account, if you have a chance to look at it, i will really appreciate it

**Comment by eriktovarj** [comment]
i have been able to send the commands but i experience pixelation, but if i watch the same stream in my computer it works with no issues.
Does it have to do with the pkt_size?

[a comment by an automated account is not shown]
